### PR TITLE
Update docs for 2.x

### LINKED ISSUE #144: Update documentation
Update the documentation to reflect the approach for version 2.

* Update Sphinx docs
* Update README
* Update examples
* Add integration documentation - e.g. django-anymail, celery, etc.

### PR DESCRIPTION
**Pull request by richleland**
Resolves #144

### CODE REVIEW

File: examples/transmissions/schedule_transmission.py [hunk 1]
```diff
@@ -2,8 +2,12 @@
 
 sp = SparkPost()
 
-response = sp.transmissions.send(
-    recipients=[
+response = sp.transmissions.post({
+    'options': {
+        'sandbox': True,
+        'start_time': '2015-11-06T09:10:00-05:00',
```
Comment (ewandennis) [inline examples/transmissions/schedule_transmission.py]: Can we show how to relate python datetime to our start_time format? It's a really common question. Something like this?

(Jeez python - include some tzinfo batteries):

```python
from datetime import datetime, timedelta, tzinfo

class EasternTime(tzinfo):
    """UTC+5 tzinfo implementation"""
    # can be configured here
    _offset = timedelta(hours = -5)
    _dst = timedelta(hours=1)
    _name = "-05:00"
    def utcoffset(self, dt):
        return self.__class__._offset
    def dst(self, dt):
        return self.__class__._dst
    def tzname(self, dt):
        return self.__class__._name

py_start_time = datetime(2015, 1, 6, 9, 10, 0, tzinfo=EasternTime())
start_time = py_start_time.strftime('%Y-%m-%dT%H:%M:%S%Z')
```

File: examples/transmissions/handle_exception.py [hunk 2]
```diff
@@ -16,12 +16,16 @@
...
+            'from': 'Testing <test@some-domain-you-havent-configured.com>',
+            'subject': 'Testing python-sparkpost exceptions',
+            'text': 'Hello world!',
+            'html': '<p>Hello world!</p>',
+        },
+        'campaign_id': 'python-sparkpost example',
+    })
 except SparkPostAPIException as err:
     # http response status code
     print(err.status)
```
Comment (ewandennis) [inline examples/transmissions/handle_exception.py]: Can we crack open the `SparkPostAPIException` type in this example? It'd be good to show how to recognise HTTP-level errors (e.g. 401 and 5xx) and extract API-level detail to help diagnose.

There's some sample code here: https://github.com/SparkPost/appengine-std-python-sample/blob/master/main.py#L32

File: README.rst [hunk 3]
```diff
@@ -74,37 +73,126 @@ Here at SparkPost, our messages are known as transmissions. Let's use the underl
...
+============
+
+Django Integration
+------------------
+
+We recommend the `django-anymail`_ package for using SparkPost with Django.
+
+.. _django-anymail: https://github.com/anymail/django-anymail
 
 Using with Google Cloud
```
Comment (richleland) [inline README.rst]: added

File: docs/index.rst [hunk 4]
```diff
@@ -33,11 +34,158 @@ Alternatively, you can pass the API key to the SparkPost class:
...
+============
+
+Django Integration
+------------------
+
+We recommend the `django-anymail`_ package for using SparkPost with Django.
+
+.. _django-anymail: https://github.com/anymail/django-anymail
+
+Using with Google Cloud
```
Comment (richleland) [inline docs/index.rst]: added

File: examples/transmissions/handle_exception.py (hunk 2 above)
Comment (richleland) [inline examples/transmissions/handle_exception.py]: added a loop that prints out error specifics

**Pull request merged**